Can't iterate Map, which was returned from a function

**V version:** V 0.1.28 bdecbc1
**OS:** Linux

**What did you do?**
```v
fn mymap() map[string]int {
  return {
    'one': 1
    'two': 2
  }
}

fn main() {
  for k, v in mymap() {
    println('$k -> $v')
  }
}
```

**What did you expect to see?**
Fine compilation.

 
**What did you see instead?**
```
/tmp/v/main.tmp.c: In function ‘main’:
/tmp/v/main.tmp.c:9548:36: error: lvalue required as unary ‘&’ operand
  array_string keys__t71 = map_keys(&mymap());
                                    ^
/tmp/v/main.tmp.c: In function ‘vcalloc’:
/tmp/v/main.tmp.c:4630:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.
```

--- a/vlib/builtin/map_test.v
+++ b/vlib/builtin/map_test.v
@@ -276,3 +276,10 @@ fn test_assign_directly() {
 	a['aaa'] -= 2
 	assert a['aaa'] == 2
 }
+
+fn test_map_in_directly() {
+	for k, v in {'aa': 1} {
+		assert k == 'aa'
+		assert v == 1
+	}
+}

--- a/vlib/v/gen/cgen.v
+++ b/vlib/v/gen/cgen.v
@@ -832,14 +832,16 @@ fn (mut g Gen) for_in(it ast.ForInStmt) {
 		idx := g.new_tmp_var()
 		key := if it.key_var in ['', '_'] { g.new_tmp_var() } else { it.key_var }
 		zero := g.type_default(it.val_type)
-		g.write('array_$key_styp $keys_tmp = map_keys(&')
+		atmp := g.new_tmp_var()
+		atmp_styp := g.typ(it.cond_type)
+		g.write('$atmp_styp $atmp = ')
 		g.expr(it.cond)
-		g.writeln(');')
+		g.writeln(';')
+		g.writeln('array_$key_styp $keys_tmp = map_keys(&$atmp);')
 		g.writeln('for (int $idx = 0; $idx < ${keys_tmp}.len; $idx++) {')
 		g.writeln('\t$key_styp $key = (($key_styp*)${keys_tmp}.data)[$idx];')
 		if it.val_var != '_' {
-			g.write('\t$val_styp ${c_name(it.val_var)} = (*($val_styp*)map_get(')
-			g.expr(it.cond)
+			g.write('\t$val_styp ${c_name(it.val_var)} = (*($val_styp*)map_get($atmp')
 			g.writeln(', $key, &($val_styp[]){ $zero }));')
 		}
 		g.stmts(it.stmts)

